segmentstore/server/src/main/java/io/pravega/segmentstore/server/DirectSegmentAccess.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server;
+
+import io.pravega.segmentstore.contracts.AttributeUpdate;
+import io.pravega.segmentstore.contracts.ReadResult;
+import io.pravega.segmentstore.contracts.SegmentProperties;
+import io.pravega.segmentstore.contracts.StreamSegmentStore;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines an API that can be used to get direct access to a Segment. This can be used instead of the SegmentContainer API
+ * for short periods of time if a rapid sequence of operations is desired (since it caches the locations of the Segment and
+ * it does not need to all the usual SegmentContainer and StreamSegment lookups on every invocation).
+ */
+public interface DirectSegmentAccess {
```

Comment:
This interface is kind of a strict subset of `StreamSegmentStore`, focusing on the operations over existing segments and eliminating the parameters that refer to the segment as the segment is given by the context of the object.

Semantically, this is not great because we are not expressing it in a way that enforces it to be a strict subset. Refactoring the interfaces would not necessarily make it any better because the signatures are different, at least I don't see a good way of doing it.

Reply:
So what is it that you are proposing? 

Please understand that the whole point of this is to:
1. Eliminate the same lookup for segment name -> container -> segment id over and over again for a rapid succession of operations working on the same segment.
2. Ease of mocking. It's much easier to mock this for testing purposes than to mock the entire container.
3. Restricting access to non-segment operations on the container. We wouldn't want someone to stop the container if they have no business doing so, would we?

Reply:
Yeah, I get the three points and agree with them. What I'm seeing is the following, and let me see if I can explain in a more clear manner.

If I check `StreamSegmentStore`, then I see for example this declaration:

```
 CompletableFuture<Void> append(String streamSegmentName, byte[] data, Collection<AttributeUpdate> attributeUpdates, Duration timeout);
```

while here I have pretty much the same declaration, except that it drops the segment name argument:

```
CompletableFuture<Long> append(byte[] data, Collection<AttributeUpdate> attributeUpdates, Duration timeout);
```

which makes sense as the direct access maps to a single segment. And, this is the case for pretty much every method in this interface, except for a couple of methods that are specific to a segment, like `getSegmentId` and `getInfo`. 

Given that it is expected that this interface consists of a strict subset of operations that are shared between `StreamSegmentStore` and `DirectSegmentAccess`, it feels like they should have a common ancestor where these common operations are declared. At the same time, I think that doing it would be messy because of the different signatures, at least I don't have any good ideas. 

Is this a better way of explaining it?